docs/docs/spec.md

```diff
@@ -726,10 +726,10 @@ A type is null-exclusive under every parameterization if it has a
 [nullness-subtype-establishing path] to either of the following:
 
 -   any type whose [nullness operator] is `MINUS_NULL`
--   any augmented class or array type
+-   any augmented class or array type or the augmented null type
 
-    > This rule refers specifically to a "class or array type," as distinct from
-    > other types like type variables and [intersection types].
+    > This rule refers to particular kinds of types as distinct from other types
```

Comment:
Maybe I'm confused by this distinction as well: doesn't `Nullness-subtype-establishing path` already make the required distinction between different types, through the `Nullness-subtype-establishing direct-supertype edges`?

Reply:
"Nullness-subtype-establishing direct-supertype edges" can include type variables, such as to support the `U`->`T` case from my previous message. To establish null-exclusivity, then, we need to reach an "actual type" (or a non-null-projected type variable). Otherwise, we'd consider `U` to be null-exclusive once we reach `T`. We instead want to look through `T` to the bound. Since the bound is `@Nullable Object`, `U` (like `T`) is _not_ null-exclusive. If it bound were `@NonNull Object`, then both type variables would be null-exclusive. (And if the bound were yet another type variable, then we'd keep looking :))